sage.py

```diff
@@ -481,88 +418,65 @@ def getepisodes(action_seq, mcat, plot, debug=False):
-    if dy[-1] < 0: # special case for last ramp down thats not fully gone down
-        negative.append((len(dy), dy[-1]))
-    elif dy[-1] > 0: # special case for last ramp up without any ramp down
-        #print('adding somthing at the end ', (len(dy), dy[-1]))
-        negative.append((len(dy), dy[-1]))
- 
-   
-    common = list(set(negative).intersection(positive))
-    negative = [item for item in negative if item not in common]
-    positive = [item for item in positive if item not in common]
-    
-    #print('--', [x[0] for x in negative] , len(y))
-    negative = [x for x in negative if (y[x[0]] <= 0 or x[0] == len(y)-1)]
-    positive = [x for x in positive if (y[x[0]] <= 0 or x[0] == 0)]
-    
-    #print(positive)
-    #print(negative)
-    
     if len(negative) < 1 or len(positive) < 1:
         return []
 
-    episodes_ = [] # Tuple (startInd, endInd)
-    for i in range(len(positive)-1):
+    # Get episodes (down between ups)
+    episodes_ = []  # Tuple (start_index, end_index)
+    for i in range(len(positive) - 1):
         ep1 = positive[i][0]
-        ep2 = positive[i+1][0]
+        ep2 = positive[i + 1][0]
         ends = []
         for j in range(len(negative)):
-
-            if negative[j][0] >= ep1 and negative[j][0] < ep2:
-                    ends.append(negative[j])
+            if ep1 <= negative[j][0] < ep2:
+                ends.append(negative[j])
         
         if len(ends) > 0:
             episode = (ep1, max([x[0] for x in ends]))
             episodes_.append(episode)
-    if (len(positive) == 1 and len(negative) == 1):
+
+    # Handle edge cases
+    if len(positive) == 1 and len(negative) == 1:
         episode = (positive[0][0], negative[0][0])
         episodes_.append(episode)
         
-    if (len(episodes_) > 0 and negative[-1][0] != episodes_[-1][1] ):  
+    if len(episodes_) > 0 and negative[-1][0] != episodes_[-1][1]:
         episode = (positive[-1][0], negative[-1][0])
         episodes_.append(episode)
      
-    
-    if (len(episodes_) > 0 and positive[-1][0] != episodes_[-1][0]):# and positive[-1][0] < episodes[-1][1]): 
+    if len(episodes_) > 0 and positive[-1][0] != episodes_[-1][0]:# and positive[-1][0] < episodes[-1][1]):
         elim = [x[0] for x in common]
         if len(elim) > 0 and max(elim) > positive[-1][0]:
             episode = (positive[-1][0], max(elim))
             episodes_.append(episode)
             
-    if (len(episodes_) == 0 and len(positive) == 2 and len(negative) == 1):
+    if len(episodes_) == 0 and len(positive) == 2 and len(negative) == 1:
         episode = (positive[1][0], negative[0][0])
         episodes_.append(episode)
-    
+
     if plot:
-        fig = plt.figure()
+        plt.figure()
         plt.title(mcat)
         plt.xlabel('Time ->')
-        plt.ylabel('Slope')
-        plt.plot(y, 'gray')
+        plt.ylabel('Frequency')
+        plt.plot(frequencies, 'gray')
```

Comment:
Can you rename the y-axis back to slope? That is more representative of what we are showing in the plot.